server/Domain/GameEvents/MetadataUpdated.cs

```diff
@@ -57,10 +58,38 @@ private MetadataUpdatedPayload Sanitize(MetadataUpdatedPayload payload)
 
         private void Validate(MetadataUpdatedPayload payload)
         {
-            // TODO TDD this stuff:
-            // TODO validate format of start and end (if and present)
-            // TODO validate if end >= start (if end present)
-            // TODO validate that VpCount > 10 && VpCount <= 14
+            if (!string.IsNullOrEmpty(payload.SessionStart))
+            {
+                DateTime dt;
+                var isValid = DateTime.TryParseExact(payload.SessionStart, "yyyy-MM-dd", new CultureInfo("en-GB"), DateTimeStyles.None, out dt);
+                if (!isValid)
+                {
+                    throw new MetadataUpdatedPayloadInvalidException("Invalid session start format");
+                }
+            }
+            
+            if (!string.IsNullOrEmpty(payload.SessionEnd))
+            {
+                DateTime dt;
+                var isValid = DateTime.TryParseExact(payload.SessionEnd, "yyyy-MM-dd", new CultureInfo("en-GB"), DateTimeStyles.None, out dt);
+                if (!isValid)
+                {
+                    throw new MetadataUpdatedPayloadInvalidException("Invalid session end format");
+                }
+            }
+            
+            if (
+                !string.IsNullOrEmpty(payload.SessionStart) 
+                && !string.IsNullOrEmpty(payload.SessionEnd) 
+                && payload.SessionStart.CompareTo(payload.SessionEnd) > 0
+            )
+                throw new MetadataUpdatedPayloadInvalidException("End should occur after start");
+
+            if (payload.VpCount < 10)
```

Comment:
I didn't agree with the `if` statement without braces, so I setup StyleCop and github action to fail with this rule

you can format the code automatically with `dotnet format` in the `./server` directory
(or setup your IDE to do it for you in whatever way you want)

please resolve conflicts and format the code and then we can merge it :)